install.sh

```diff
@@ -1,90 +1,355 @@
+#! /usr/bin/env bash
+set -uo pipefail
+
+# Setting up color and style variables
+bold=$(tput bold)
+blue=$(tput setaf 4)
+green=$(tput setaf 2)
+yellow=$(tput setaf 3)
+red=$(tput setaf 1)
+reset=$(tput sgr0)
+white=$(tput setaf 7)
+term_width=$(tput cols)
+box_width=60
+padding="$(printf '%*s' $(( (term_width - box_width) / 2 )) '')"
+box_line="$(printf '%*s' $box_width '')"
+box_line="${box_line// /-}"
+
+# Function to display error messages in red
+display_error() {
+    echo "${bold}${red}$1${reset}"
+}
+
+# Checking for AWS credentials
+if [[ -z "$AWS_ACCESS_KEY_ID" || -z "$AWS_SECRET_ACCESS_KEY" || -z "$AWS_SESSION_TOKEN" ]]; then
+    display_error "Missing AWS credentials. Please configure the AWS CLI with your credentials."
+    exit 1
+else
+    echo "${bold}${green}AWS credentials detected successfully.${reset}"
+fi
+
+# Trying to retrieve AWS account owner's details
+if ! AWS_ACCOUNT_DETAILS_JSON=$(aws sts get-caller-identity 2>&1); then
+    display_error "Unable to obtain AWS caller identity: $AWS_ACCOUNT_DETAILS_JSON"
+    display_error "Check if your AWS credentials are expired and you have appropriate permissions."
+    exit 1
+fi
+
+# Extracting and displaying account details
+AWS_ACCOUNT_ID=$(echo "$AWS_ACCOUNT_DETAILS_JSON" | jq -r '.Account')
+AWS_USER_ID=$(echo "$AWS_ACCOUNT_DETAILS_JSON" | jq -r '.UserId')
+AWS_ARN=$(echo "$AWS_ACCOUNT_DETAILS_JSON" | jq -r '.Arn')
+AWS_ROLE=$(aws sts get-caller-identity --query 'Arn' --output text | cut -d '/' -f 2)
+
+# Function to print a line with padding
+print_line() {
+    echo "${padding}${blue}${white}$1${reset}"
+}
+
+# Displaying AWS account information in a "box"
+echo "${padding}${box_line}"
+echo
+print_line "${bold}AWS Account Information:${reset}"
+echo
+print_line "Account ID: ${bold}$AWS_ACCOUNT_ID${reset}"
+print_line "User ID: ${bold}$AWS_USER_ID${reset}"
+print_line "Role: ${bold}$AWS_ROLE${reset}"
+echo
+echo "${padding}${box_line}"
+
+
+echo
+# Ask consent to proceed with the aws account
+while true; do
+    read -r -p "Do you want to proceed with the above AWS account? [y/n]: " yn
```

Comment:
we need to bypass this in single-click deployment as it happens within an ec2-instance

Reply:
Yes, we will be using a seperate simple single-click-install.sh that'll bypass most of the stuff which is not required.